lib/vmpooler/api/v1.rb

```diff
@@ -764,6 +932,78 @@ def invalid_pool(payload)
       invalid
     end
 
+    def check_ondemand_request(request_id)
+      result = { 'ok' => false }
+      request_hash = backend.hgetall("vmpooler__odrequest__#{request_id}")
+      if request_hash.empty?
+        result['message'] = "no request found for request_id '#{request_id}'"
+        return result
+      end
+
+      result['request_id'] = request_id
+      result['ready'] = false
+      result['ok'] = true
+      status 202
+
+      if request_hash['status'] == 'ready'
+        result['ready'] = true
+        platform_parts = request_hash['requested'].split(',')
```

Comment:
```suggestion
        platform_parts = request_hash['requested'].split(',')
        platform_parts.each do |platform|
          pool_alias, pool, _count = platform.split(':')
          instances = backend.smembers("vmpooler__#{request_id}__#{pool_alias}__#{pool}")
```

I saw variations of this in 4 places, where we deconstruct I wonder if a method that accepts a block could DRY it? And we could add some input data validation around the `split()` I'm thinking what if `split(":")` has nil for `pool_alias`, `pool`
Is it possible for `backend.smembers` to also return nil?

Reply:
2. https://github.com/puppetlabs/vmpooler/pull/375/files#diff-51d4606238cfb066704749b609c2d728R965
3. https://github.com/puppetlabs/vmpooler/pull/375/files#diff-51d4606238cfb066704749b609c2d728R993
4. https://github.com/puppetlabs/vmpooler/pull/375/files#diff-9ca942eee48c471766a71aa4fd1df851R1477
5. https://github.com/puppetlabs/vmpooler/pull/375/files#diff-9ca942eee48c471766a71aa4fd1df851R1523

Reply:
Maybe a method that handles grabbing the string, splitting it up and returning the vars to be assigned?
`backend.smembers` returns an empty array when the set is empty.

Reply:
I looked at this a little more and am not quite sure the best way to address this. We could have a method return the pool_alias, pool and count for each of the requested platforms in the requested string, and then take the action on that returned data. I'm not sure if it clarifies things much.
I didn't see a  clear  way  to make this better with a block. Did you have a specific idea  in mind for how to improve this? I'm open to ideas, but did  not  see  a clear path to improvement.

Reply:
yes a utility method like
```
    def split_platform_pool_count(requested, &block)
      requested_platforms = requested.split(',')
        requested_platforms.each do |platform|
          platform_alias, pool, count = platform.split(':')
         #TODO data validation on platform_alias, pool, count if nil
          yield platform_alias, pool, count
        end
    end
```
could then be called in the first example like:
```
split_platform_pool_count(request_hash['requested']) do |platform_alias,pool,count|
            instances = backend.smembers("vmpooler__#{request_id}__#{pool_alias}__#{pool}")
          result[pool_alias] = { 'hostname': instances }
end
```
the second example in manager becomes
```
split_platform_pool_count(request_hash['requested']) do |platform_alias,pool,count|
  platform_alias, pool, count = platform.split(':')
  pools_filled = redis.scard("vmpooler__#{request_id}__#{platform_alias}__#{pool}")
  throw :request_not_ready unless pools_filled.to_i == count.to_i
end
```
etc

obviously naming can be improved and do we have a natural place for that utility method?

Reply:
We do not. API has its own helpers class. So without introducing something new, like a generic helpers class, I would have to duplicate the utility method in API and manager.